pkg/ebpf/encoding/encoding_test.go

```diff
@@ -0,0 +1,114 @@
+package encoding
+
+import (
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/ebpf/netlink"
+	agent "github.com/DataDog/datadog-agent/pkg/process/model"
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSerialization(t *testing.T) {
+	in := &ebpf.Connections{
+		Conns: []ebpf.ConnectionStats{
+			{
+				Source:               util.AddressFromString("10.1.1.1"),
+				Dest:                 util.AddressFromString("10.2.2.2"),
+				MonotonicSentBytes:   1,
+				LastSentBytes:        2,
+				MonotonicRecvBytes:   100,
+				LastRecvBytes:        101,
+				LastUpdateEpoch:      50,
+				MonotonicRetransmits: 201,
+				LastRetransmits:      201,
+				Pid:                  6000,
+				NetNS:                7,
+				SPort:                1000,
+				DPort:                9000,
+				IPTranslation: &netlink.IPTranslation{
+					ReplSrcIP:   "20.1.1.1",
+					ReplDstIP:   "20.1.1.1",
+					ReplSrcPort: 40,
+					ReplDstPort: 70,
+				},
+
+				Type:      ebpf.UDP,
+				Family:    ebpf.AFINET6,
+				Direction: ebpf.LOCAL,
+			},
+		},
+	}
+
+	out := &agent.Connections{
+		Conns: []*agent.Connection{
+			{
+				Laddr:              &agent.Addr{Ip: "10.1.1.1", Port: int32(1000)},
+				Raddr:              &agent.Addr{Ip: "10.2.2.2", Port: int32(9000)},
+				TotalBytesSent:     1,
+				LastBytesSent:      2,
+				TotalBytesReceived: 100,
+				LastBytesReceived:  101,
+				TotalRetransmits:   201,
+				LastRetransmits:    201,
+				Pid:                int32(6000),
+				NetNS:              7,
+				IpTranslation: &agent.IPTranslation{
+					ReplSrcIP:   "20.1.1.1",
+					ReplDstIP:   "20.1.1.1",
+					ReplSrcPort: int32(40),
+					ReplDstPort: int32(70),
+				},
+
+				Type:      agent.ConnectionType_udp,
+				Family:    agent.ConnectionFamily_v6,
+				Direction: agent.ConnectionDirection_local,
+			},
+		},
+	}
+
+	t.Run("requesting application/json serialization", func(t *testing.T) {
+		assert := assert.New(t)
+		marshaler := GetMarshaler("application/json")
+		assert.Equal("application/json", marshaler.ContentType())
+
+		blob, err := marshaler.Marshal(in)
+		assert.Nil(err)
```

Comment:
Small nit, we typically use `require` for errors so that the test halts immediately.  Using `assert` will allow the test to proceed until the end which may cause weird panics depending on what happens after the error

Reply:
Didn't know about that. I'll update it.